### ISSUE TITLE
Possible future feature - in-call spam reporting

### ISSUE THREAD

**Issue by lgaetz** [FreePBX/superfecta#205]
A bit of far out brainstorming, what if ...
- Superfecta creates a new in-call feature code such as "_7726" (_SPAM) that is dtmf input during a live call, sets channel variables such as `${SPAM_NUMBER}` and `${SPAM_NAME}` (I assume in-call feature codes can be created by a module?)
- create a new function within the individual lookup module classes called `post_call()` that receives `${SPAM_NUMBER}` and `${SPAM_NAME}` as arguments. The code in this function would execute after a call is finished.
- Superfecta creates a context that runs at call completion that checks the value of these variables, and if set, executes the `post_call()` functions in the individual module classes.
- Checks the value of Asterisk as ver 11+ and adds a hanguphandler for all inbound calls to the context created above.

The gist is that the new in-call feature code gives the user the ability to effortlessly report a spam caller while they are on the line. I am sure there would be some possibly major structural changes required to do this, maybe there is a better way to accomplish the same end with existing superfecta code.

**Comment by tm1000** [comment]
Yes in call feature codes can execute any dialplan, so external scripts and such are acceptable. Once you hit the feature code it is run instantly, not at the end of the call.
